src/backends/backend.js

```diff
@@ -70,8 +71,11 @@ class Backend {
   currentUser() {
     if (this.user) { return this.user; }
     const stored = this.authStore && this.authStore.retrieve();
-    if (stored) {
-      return Promise.resolve(this.implementation.setUser(stored)).then(() => stored);
+    if (stored && (stored.backend === this.backendName)) {
+      return this.implementation.restoreUser(stored).then((user) => {
```

Comment:
@Benaiah @erquhart What do you think is the best way to do this (especially for the `git-gateway` backend)? Completely remove the `restoreUser` function and send the stored user to `authenticate`, or keep it and just forward the calls for most backends like I did here?

Reply:
Removing the `restoreUser` function would mean users always have to log back in, no?

Reply:
If removing `restoreUser` requires the user to log back in again anyway I'd say definitely leave this as is. Defining `restoreUser` as a call to `authenticate` in the backend seems like a perfectly fine way to handle backends that can't do any other restoration.

Reply:
Good, I'll leave it like it is, then.

Reply:
@erquhart No, I was mostly talking about the forwarding part of it, which @Benaiah addressed.